python_modules/dagster/dagster/_core/blueprints/load_from_yaml.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+from typing import Type, Union
+
+from dagster import (
+    Definitions,
+    _check as check,
+)
+from dagster._utils.pydantic_yaml import parse_yaml_file
+
+from .blueprint import Blueprint
+
+
+def load_defs_from_yaml(
+    *, path: Union[Path, str], per_file_blueprint_type: Type[Blueprint]
+) -> Definitions:
+    """Load Dagster definitions from a YAML file of blueprints.
+
+    Args:
+        path (Path | str): The path to the YAML file or directory of YAML files containing the
+            blueprints for Dagster definitions.
+        per_file_blueprint_type (type[Blueprint]): The type of blueprint that each of the YAML
```

Comment:
I want to probe a bit at the decision to specify the blueprint type in Python code - how would we suggest users implement some sort of "union type" where a file could define one of a few types of assets?

I could imagine a few ways to do this:
- Separate `load_defs_from_yaml` calls targeting each set of files
- A wrapper `Blueprint` type with a `Selector`/union type at the top level. This means you need to have some type which imports/explicitly unions all possible blueprints


I think it's a reasonable API decision but I do find the other option - letting the user encode the blueprint type in the YAML file - also pretty nice:

```yaml
_blueprint:
  module: my_module.blueprints
  class: MyFooBlueprint

a_str: foo
an_int: 55
```

This is less elegant, but avoids user code having to know about all possible blueprints, or defining a big wrapper type. This would be useful for things like integration-provided blueprints:

```yaml
_blueprint:
  module: dagster_embedded_elt.sling
  class: SlingSyncBlueprint

...
```

```yaml
_blueprint:
  module: dagster_dbt
  class: DbtModelBlueprint

...
```

That way a user can just call

`defs = load_defs_from_yaml(...)`

With a dir full of various YAML files that point at different integrations, as opposed to multiple calls which have to filter out each match/non-match.

Reply:
There's also a world in which specifying blueprint-in-yaml could nicely match syntactically with [linking a YAML schema](https://github.com/redhat-developer/vscode-yaml?tab=readme-ov-file#associating-schemas) for editor type checking:

```yaml
# dagster: blueprint=dagster_embedded_elt.sling.SlingSyncBlueprint
# yaml-language-server: $schema=./.schemas/dagster_embedded_elt.sling.SlingSyncBlueprint
```

Reply:
>  avoids user code having to know about all possible blueprints

The way I have been thinking about it, this is a feature more than a bug.  I.e. a big goal of blueprints is to help platform owners create guardrails for their stakeholders. By defining the set of available blueprints up front, you can be confident that only those will show up in the loaded definitions.

Reply:
> I think it makes sense that this is Type[Blueprint] at least right now. In the future it might be ergonomic to allow for for e.g. List[Blueprint] (which would just have a special cased build_defs impl, I guess).

Agreed. I actually started implementing this in this PR, but it added enough complexity that I figured it would be better to consider it separately. Same with unions.